src/decode.rs

```diff
@@ -790,14 +791,20 @@ unsafe fn read_pal_indices(
             pal_tmp[offset..][..len].fill(filler);
         }
     }
-    (pal_dsp.pal_idx_finish)(
-        pal_idx.unwrap_or(pal_tmp).as_mut_ptr(),
-        pal_tmp.as_ptr(),
-        bw4 as c_int * 4,
-        bh4 as c_int * 4,
-        w4 as c_int * 4,
-        h4 as c_int * 4,
-    );
+
+    // SAFETY: Unsafe asm call, checkasm tests have verified that the signature is
+    // correct. `pal_idx` and `pal_tmp` are at least `(bw4 * 4) * (bh4 * 4)`
+    // elements long, which is how many elements `pal_idx_finish` will read/write.
```

Comment:
These kinds of assertions should be `debug_assert!`s at the very least, I think.

Reply:
Added. I also tweaked the wording of the safety comment since the required length of the two arrays is slightly different.